moped-editor/src/views/projects/projectView/ProjectComponents/RelatedProjectSourcesAndLayers.js

```diff
@@ -0,0 +1,96 @@
+import { Source, Layer } from "react-map-gl";
+import { MAP_STYLES } from "./mapStyleSettings";
+import { useFeatureTypes } from "./utils/map";
+
+/**
+ * Component that renders feature collection of related project component features
+ * All layers are set to show below basemap street labels using beforeId = "street-labels"
+ * @param {Boolean} isCreatingComponent - are we creating a component?
+ * @param {Boolean} isEditingComponent - are we editing a component?
+ * @param {Object} featureCollection - GeoJSON feature collection with all project features
+ * @param {Boolean} shouldShowRelatedProjects - should we show related projects
+ * @param {Object} clickedComponent - Details of the component that was clicked
+ * @returns JSX.Element
+ */
+const RelatedProjectSourcesAndLayers = ({
+  isCreatingComponent,
+  isEditingComponent,
+  featureCollection,
+  shouldShowRelatedProjects,
+  clickedComponent,
+}) => {
+  const projectLines = useFeatureTypes(featureCollection, "line");
+  const projectPoints = useFeatureTypes(featureCollection, "point");
+
+  const isCreatingOrEditing = isCreatingComponent || isEditingComponent;
+  const shouldShowNonMutedFeatures =
+    shouldShowRelatedProjects && !clickedComponent && !isCreatingOrEditing;
+  const shouldShowMutedFeatures =
+    (shouldShowRelatedProjects && !!clickedComponent) ||
+    (shouldShowRelatedProjects && isCreatingOrEditing);
```

Comment:
I gotta call out how great these variable names are, makes it easier for me to follow whats going on